From: SeongJae Park <sj@kernel.org>
Date: Sat, 14 Jun 2025 14:05:20 -0700
Subject: [PATCH] ==== lru_sort advancing ====

mm/damon: advance DAMOS-based LRU sorting

DAMOS_LRU_[DE]PRIO actions were added to DAMOS for more access-aware LRU
lists sorting.  For simple usage, a specialized kernel module, namely
DAMON_LRU_SORT, has also introduced.  After the introduction of
DAMON_LRU_SORT, DAMON got a few important new features, including the
aim-based quota auto-tuning, young page filter, and monitoring intervals
auto-tuning.  But DAMOS-based LRU sorting had no many updates.  Now we
show some rooms to advance for DAMOS-based LRU sorting.

DAMON sysfs interface users can utilize all the features, but
DAMON_LRU_SORT is not updated for such modern features.  Also, even in
case of DAMON sysfs interface usage, there is no good quota auto-tuning
goal metric for LRU sorting.  Since it is a sort of balancing of active
and inactive pages, a metric that represents the size ratio between the
active and the inactive pages could be useful.  Also, internal behaviors
of DAMOS_LRU_[DE]PRIO are not very symmetric.

Advance DAMOS-based LRU sorting by introducing a new DAMOS quota
auto-tuning target metric for [in]active memory size ratio, making
DAMOS_LRU_[DE]PRIO behavior more intuitive and symmetric.  Also update
DAMON_LRU_SORT module to be able to fully utilize the modern features
including the [in]active memory size ratio, young page filter, and
monitoring intervals auto-tuning.

With these changes, for example, users can now ask DAMON to "find
hot/cold memory regions with auto-tuned monitoring intervals, do one
more page level access check for found hot/cold memory, and move pages
of those to active or inactive LRU lists accordingly, aiming 50:50
active to inactive memory ratio."

Plan for none-RFC Version
-------------------------

I'm planning to do more tests and add the results by the final version.

Patches Sequence
----------------

First four patches extend DAMOS quota auto-tuning to support [in]active
memory ratio target metric type.  The patches introduce new metrics
(patch 1), DAMON sysfs support
(patch 2), and add the document (patch 3).

Following patch (patch 4) makes DAMOS_LRU_PRIO action to directly move
target pages to active LRU list, instead of marking accessed and
incrementally be moved to active lists.

Following seven patches (patches 5-11) updates DAMON_LRU_SORT to support
modern DAMON features.  Patch 5 makes it to use not only access
frequency but also age at under-quota regions prioritization.  Patch
6-11 add the support and document it for young page filtering,
active:inactive memory ratio based quota auto-tuning, and monitoring
intervals auto-tuning, respectively.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 hkml_cv_bogus/hkml_cv_bogus_mv9z21za | 0
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 hkml_cv_bogus/hkml_cv_bogus_mv9z21za

diff --git a/hkml_cv_bogus/hkml_cv_bogus_mv9z21za b/hkml_cv_bogus/hkml_cv_bogus_mv9z21za
new file mode 100644
index 000000000000..e69de29bb2d1
-- 
2.39.5

